src/RemoteStorageForm/RemoteStorageForm.js

```diff
@@ -63,13 +63,12 @@ const RemoteStorageForm = ({
   ), [intl, handleSubmit, pristine, submitting, onClose]);
 
   const changeProvider = useCallback(({ values }) => {
-    if (values?.providerName === 'DEMATIC_SD') {
-      setIsDematicSD(true);
-    } else {
-      setIsDematicSD(false);
-    }
+    setSelectedProviter(values.providerName);
   }, []);
 
+  const isDematicSD = useMemo(() => selectedProvider === 'DEMATIC_SD', [selectedProvider]);
```

Comment:
hmm. is it so expensive operation? It's just an equality and result is a boolean. IMHO, it's overkill - `useMemo` here